reference/src/representation/enums.md

```diff
@@ -0,0 +1,376 @@
+# Representation of Rust `enum` types
+
+**Disclaimer:** Some parts of this section were decided in RFCs, but
+others represent the consensus from issue [#10]. The text will attempt
+to clarify which parts are "guaranteed" (owing to the RFC decision)
+and which parts are still in a "preliminary" state.
+
+[#10]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/10
+
+## Background
+
+**C-like enums.** The simplest form of enum is simply a list of
```

Comment:
These are called field-less enums nowadays (see e.g. [the reference](https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-field-less-enumerations)).